data/data/aws/variables-aws.tf

```diff
@@ -87,6 +87,11 @@ variable "aws_private_subnets" {
   description = "(optional) Existing private subnets into which the cluster should be installed."
 }
 
+variable "aws_public_zone_id" {
+  type = string
+  description = "The ID of the public DNS zone"
+}
```

Comment:
in case of internal only clusters, the installer will pass empty public zone ID, and the terraform will error out with no public zone id provided?

Reply:
The only place the zone id is actually used is here:
<pre>
data "aws_route53_zone" "public" {
  count = local.public_endpoints ? 1 : 0

  zone_id = var.public_zone_id
}
</pre>

Isn't local_endpoints set when this is an internal cluster?

Reply:
I'm not sure your comment answers my internal cluster's concern, maybe I'm wrong. can you verify your code works with internal clusters.

Reply:
I have confirmed this works for internal clusters.